Classes/DDLog.m

```diff
@@ -1107,15 +1107,16 @@ - (instancetype)initWithMessage:(NSString *)message
                         options:(DDLogMessageOptions)options
                       timestamp:(NSDate *)timestamp {
     if ((self = [super init])) {
-        _message      = [message copy];
+        BOOL copyMessage = (options & DDLogMessageDontCopyMessage) == 0;
+        _message      = copyMessage ? [message copy] : message;
```

Comment:
IMHO, I first thought that the compiler was smart enough to produce to the same code. But since I wanted to be sure, I double-checked by using a disassembler, and guess what? The code is the same! You can verify it by looking at the attachments, line 49-51: the first file is with the explicit variable, the second is your version.
For my part, I prefer the first version since it looks like the code already there.

[iOSdm Daemon_-[DDLogMessage initWithMessage:level:flag:context:file:function:line:tag:options:timestamp:].txt](https://github.com/CocoaLumberjack/CocoaLumberjack/files/771654/iOSdm.Daemon_-.DDLogMessage.initWithMessage.level.flag.context.file.function.line.tag.options.timestamp.txt)
[iOSdm Daemon_-[DDLogMessage initWithMessage:level:flag:context:file:function:line:tag:options:timestamp:]_2.txt](https://github.com/CocoaLumberjack/CocoaLumberjack/files/771655/iOSdm.Daemon_-.DDLogMessage.initWithMessage.level.flag.context.file.function.line.tag.options.timestamp._2.txt)